src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -620,7 +622,7 @@ public byte[] produce(ConnectionContext context,
 
             SSLKeyDerivation handshakeKD = ke.createKeyDerivation(shc);
             SecretKey handshakeSecret = handshakeKD.deriveKey(
```

Comment:
It looks like this can be cleared after it is used to derive the key. Similar comment on line 1310.

Reply:
Well, I am not sure if clearing `handshakeSecret` is ok - this `handshakeSecret` is passed to `kd` on line 636 and stored internally without cloning. Then `kd` is stored into `shc` which suggests that it may be used later. Clearing it will likely cause problems for subsequent key derivations? Same goes for line 1310. Is there something that I missed?

Reply:
Ah, yes you are right.